src/resource.c

```diff
@@ -995,6 +1004,40 @@ coap_resource_notify_observers(coap_resource_t *r, const coap_string_t *query) {
   return 1;
 }
 
+void
+coap_resource_set_mode(coap_resource_t *resource, int mode) {
+  resource->flags = (resource->flags &
+    ~(COAP_RESOURCE_FLAGS_NOTIFY_CON|COAP_RESOURCE_FLAGS_NOTIFY_NON)) |
+    (mode & (COAP_RESOURCE_FLAGS_NOTIFY_CON|COAP_RESOURCE_FLAGS_NOTIFY_NON));
+}
+
+void
+coap_resource_set_userdata(coap_resource_t *resource, void *data) {
+  resource->user_data = data;
```

Comment:
Maybe call `release_userdata` first so that `coap_resource_set_userdata()` can be called to safely replace existing userdata?

Reply:
I like the principal, but this would cause things to blow up - as per `examples/etsi_iot_01.c` (where direct access to resource was replaced by functions).
````
static inline void
coap_delete_payload(coap_resource_t *resource) {
  coap_free(coap_resource_get_userdata(resource));
  coap_resource_set_userdata(resource, NULL);
}
````

Reply:
Okay, this makes sense.